appIcons.js

```diff
@@ -166,6 +166,11 @@ const MyAppIcon = new Lang.Class({
 
     _updateRunningStyle: function() {
         this.parent();
+        if (this._dtdSettings.get_boolean('isolate-workspaces')) {
```

Comment:
Ok. Can you please add he explanation of what we are doing with a comment? 

Also, to be more explicit I think this should be (please check the code is correct):

```
if (this._dtdSettings.get_boolean('isolate-workspaces')) {
        if (this.app.state != Shell.AppState.STOPPED && this.app.is_on_workspace(global.screen.get_active_workspace())
            this._dot.show();
        else
            this._dot.hide();
        this._onFocusAppChanged();
}
else {
        this.parent();
}